@spwoodcock one last obstacle remaining in front of us:

```
======================================================================
FAIL: test_duplicate_email_returns_400 (qfieldcloud.core.tests.test_user.CreatePersonAPITestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/app/qfieldcloud/core/tests/test_user.py", line 490, in test_duplicate_email_returns_400
    self.assertIn("email", response.data)
AssertionError: 'email' not found in {'code': 'api_error', 'message': 'API Error'}

----------------------------------------------------------------------```

--- a/docker-app/qfieldcloud/core/permissions_utils.py
+++ b/docker-app/qfieldcloud/core/permissions_utils.py
@@ -541,6 +541,15 @@ def can_list_users_organizations(user: QfcUser) -> bool:
     return True
 
 
+def can_create_user(user: QfcUser) -> bool:
+    """Return True if the user can create new QFieldCloud accounts.
+
+    Restricted to staff so only service accounts and admins can provision
+    users programmatically via the API.
+    """
+    return user.is_staff
+
+
 def can_create_organizations(user: QfcUser) -> bool:
     return user.is_authenticated
 

--- a/docker-app/qfieldcloud/core/serializers.py
+++ b/docker-app/qfieldcloud/core/serializers.py
@@ -19,6 +19,7 @@ from qfieldcloud.core.models import (
     Organization,
     OrganizationMember,
     PackageJob,
+    Person,
     ProcessProjectfileJob,
     Project,
     ProjectCollaborator,
@@ -266,6 +267,36 @@ class CompleteUserSerializer(serializers.ModelSerializer):
             "last_name",
         )
         read_only_fields = ("full_name", "avatar_url")
+
+
+class CreateUserSerializer(serializers.Serializer):
+    """Serializer for programmatic Person account creation.
+
+    Intentionally does *not* validate username uniqueness here so that the view
+    can return HTTP 409 Conflict (rather than 400) when the username is taken.
+    """
+
+    username = serializers.RegexField(
+        r"^[-a-zA-Z0-9_]+$",
+        max_length=150,
+        min_length=3,
+        error_messages={
+            "invalid": _(
+                "Enter a valid username. This value may contain only letters, "
+                "numbers, and -/_ characters."
+            )
+        },
+    )
+    password = serializers.CharField(write_only=True, style={"input_type": "password"})
+    email = serializers.EmailField(required=True)
+
+    def create(self, validated_data):
+        return Person.objects.create_user(
+            username=validated_data["username"],
+            password=validated_data["password"],
+            email=validated_data["email"],
+            has_accepted_tos=True,
+        )
 
 
 class PublicInfoUserSerializer(serializers.ModelSerializer):

--- a/docker-app/qfieldcloud/core/serializers.py
+++ b/docker-app/qfieldcloud/core/serializers.py
@@ -276,25 +276,22 @@ class CreateUserSerializer(serializers.Serializer):
     can return HTTP 409 Conflict (rather than 400) when the username is taken.
     """
 
-    username = serializers.RegexField(
-        r"^[-a-zA-Z0-9_]+$",
+    username = serializers.CharField(
         max_length=150,
-        min_length=3,
-        error_messages={
-            "invalid": _(
-                "Enter a valid username. This value may contain only letters, "
-                "numbers, and -/_ characters."
-            )
-        },
+        validators=User._meta.get_field("username").validators,
     )
     password = serializers.CharField(write_only=True, style={"input_type": "password"})
     email = serializers.EmailField(required=True)
+    first_name = serializers.CharField(max_length=150, required=False, default="")
+    last_name = serializers.CharField(max_length=150, required=False, default="")
 
     def create(self, validated_data):
         return Person.objects.create_user(
             username=validated_data["username"],
             password=validated_data["password"],
             email=validated_data["email"],
+            first_name=validated_data["first_name"],
+            last_name=validated_data["last_name"],
             has_accepted_tos=False,
         )
 

--- a/docker-app/qfieldcloud/core/tests/test_user.py
+++ b/docker-app/qfieldcloud/core/tests/test_user.py
@@ -473,6 +473,22 @@ class CreatePersonAPITestCase(APITestCase):
         )
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
+    def test_duplicate_email_returns_400(self):
+        Person.objects.create_user(
+            username="emailowner", password="abc123", email="taken@example.com"
+        )
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.post(
+            self._url(),
+            {
+                "username": "newuser",
+                "password": "abc123",
+                "email": "taken@example.com",
+            },
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("email", response.data)
+
     def test_list_still_works_for_staff(self):
         """GET /api/v1/users/ must still function after adding POST support."""
         self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)

--- a/docker-app/qfieldcloud/core/tests/test_user.py
+++ b/docker-app/qfieldcloud/core/tests/test_user.py
@@ -326,3 +326,126 @@ class QfcTestCase(APITestCase):
             Person.objects.create_user(
                 username="u2", password="abc123", email="same@example.com"
             )
+
+
+class CreatePersonAPITestCase(APITestCase):
+    """Tests for POST /api/v1/users/ (staff-only)."""
+
+    def setUp(self):
+        setup_subscription_plans()
+
+        self.staff_user = Person.objects.create_user(
+            username="staffuser",
+            password="abc123",
+            email="staff@example.com",
+            is_staff=True,
+        )
+        self.staff_token = AuthToken.objects.get_or_create(
+            user=self.staff_user,
+            client_type=AuthToken.ClientType.QFIELD,
+            user_agent="qfield|dev",
+        )[0]
+
+        self.regular_user = Person.objects.create_user(
+            username="regularuser",
+            password="abc123",
+            email="regular@example.com",
+        )
+        self.regular_token = AuthToken.objects.get_or_create(
+            user=self.regular_user,
+            client_type=AuthToken.ClientType.QFIELD,
+            user_agent="qfield|dev",
+        )[0]
+
+        set_subscription((self.staff_user, self.regular_user), "default_user")
+
+    def _url(self):
+        return "/api/v1/users/"
+
+    def test_staff_can_create_user(self):
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.post(
+            self._url(),
+            {
+                "username": "newmapper",
+                "password": "securepass123",
+                "email": "newmapper@example.com",
+            },
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data["username"], "newmapper")
+        self.assertTrue(Person.objects.filter(username="newmapper").exists())
+
+    def test_create_person_with_email(self):
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.post(
+            self._url(),
+            {"username": "newmapper2", "password": "pass", "email": "m@example.com"},
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(
+            Person.objects.get(username="newmapper2").email, "m@example.com"
+        )
+
+    def test_email_required(self):
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.post(
+            self._url(), {"username": "nomailuser", "password": "pass"}
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_create_person_sets_has_accepted_tos(self):
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        self.client.post(
+            self._url(),
+            {"username": "tosuser", "password": "pass", "email": "tos@example.com"},
+        )
+        self.assertTrue(Person.objects.get(username="tosuser").has_accepted_tos)
+
+    def test_duplicate_username_returns_409(self):
+        Person.objects.create_user(username="existing", password="abc123")
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.post(
+            self._url(),
+            {
+                "username": "existing",
+                "password": "pass",
+                "email": "existing@example.com",
+            },
+        )
+        self.assertEqual(response.status_code, status.HTTP_409_CONFLICT)
+
+    def test_non_staff_forbidden(self):
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.regular_token.key)
+        response = self.client.post(
+            self._url(), {"username": "shouldfail", "password": "pass"}
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_unauthenticated_rejected(self):
+        response = self.client.post(
+            self._url(), {"username": "anon", "password": "pass"}
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_invalid_username_rejected(self):
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.post(
+            self._url(),
+            {"username": "bad username!", "password": "pass", "email": "m@example.com"},
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_username_too_short_rejected(self):
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.post(
+            self._url(),
+            {"username": "ab", "password": "pass", "email": "m@example.com"},
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_list_still_works_for_staff(self):
+        """GET /api/v1/users/ must still function after adding POST support."""
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.staff_token.key)
+        response = self.client.get(self._url())
+        self.assertEqual(response.status_code, status.HTTP_200_OK)

--- a/docker-app/qfieldcloud/core/urls.py
+++ b/docker-app/qfieldcloud/core/urls.py
@@ -48,7 +48,7 @@ urlpatterns = [
     *subscription_urlpatterns,
     path("projects/public/", projects_views.PublicProjectsListView.as_view()),
     path("", include(router.urls)),
-    path("users/", users_views.ListUsersView.as_view()),
+    path("users/", users_views.ListCreateUsersView.as_view()),
     path(
         "users/<str:username>/organizations/",
         users_views.ListUserOrganizationsView.as_view(),

--- a/docker-app/qfieldcloud/core/views/users_views.py
+++ b/docker-app/qfieldcloud/core/views/users_views.py
@@ -1,30 +1,36 @@
 from django.contrib.auth import get_user_model
 from django.core.exceptions import ObjectDoesNotExist
+from django.db import IntegrityError
 from drf_spectacular.utils import extend_schema, extend_schema_view
 from qfieldcloud.core import pagination, permissions_utils, querysets_utils
 from qfieldcloud.core.models import Organization, Project
 from qfieldcloud.core.serializers import (
     CompleteUserSerializer,
+    CreateUserSerializer,
     OrganizationSerializer,
     PublicInfoUserSerializer,
 )
-from rest_framework import generics, permissions
+from rest_framework import generics, permissions, status
 from rest_framework.response import Response
 
 User = get_user_model()
 
 
-class ListUsersViewPermissions(permissions.BasePermission):
+class ListCreateUsersViewPermissions(permissions.BasePermission):
     def has_permission(self, request, view):
+        if request.method == "POST":
+            return permissions_utils.can_create_user(request.user)
+
         return permissions_utils.can_list_users_organizations(request.user)
 
 
 @extend_schema_view(
     get=extend_schema(description="List users and/or organizations"),
+    post=extend_schema(description="Create a new user account (staff only)"),
 )
-class ListUsersView(generics.ListAPIView):
+class ListCreateUsersView(generics.ListCreateAPIView):
     serializer_class = PublicInfoUserSerializer
-    permission_classes = [permissions.IsAuthenticated, ListUsersViewPermissions]
+    permission_classes = [permissions.IsAuthenticated, ListCreateUsersViewPermissions]
     pagination_class = pagination.QfcLimitOffsetPagination()
 
     def get_queryset(self):
@@ -60,6 +66,22 @@ class ListUsersView(generics.ListAPIView):
             exclude_organizations=exclude_organizations,
             exclude_teams=exclude_teams,
             invert=invert,
+        )
+
+    def post(self, request):
+        serializer = CreateUserSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            user = serializer.save()
+        except IntegrityError:
+            return Response(
+                {"username": ["A user with this username already exists."]},
+                status=status.HTTP_409_CONFLICT,
+            )
+
+        return Response(
+            PublicInfoUserSerializer(user).data,
+            status=status.HTTP_201_CREATED,
         )
 
 

